libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/AudioCapabilities.java

```diff
@@ -102,25 +100,33 @@ public static AudioCapabilities getCapabilities(Context context) {
     if (Util.SDK_INT >= 23 && Api23.isBluetoothConnected(context)) {
       return DEFAULT_AUDIO_CAPABILITIES;
     }
+
+    ImmutableSet.Builder supportedEncodings = new ImmutableSet.Builder<>();
     if (deviceMaySetExternalSurroundSoundGlobalSetting()
         && Global.getInt(context.getContentResolver(), EXTERNAL_SURROUND_SOUND_KEY, 0) == 1) {
-      return EXTERNAL_SURROUND_SOUND_CAPABILITIES;
+      supportedEncodings.addAll(Ints.asList(EXTERNAL_SURROUND_SOUND_ENCODINGS));
     }
     // AudioTrack.isDirectPlaybackSupported returns true for encodings that are supported for audio
     // offload, as well as for encodings we want to list for passthrough mode. Therefore we only use
     // it on TV and automotive devices, which generally shouldn't support audio offload for surround
     // encodings.
     if (Util.SDK_INT >= 29 && (Util.isTv(context) || Util.isAutomotive(context))) {
+      supportedEncodings.addAll(Ints.asList(Api29.getDirectPlaybackSupportedEncodings()));
+    }
+
+    if (intent == null || intent.getIntExtra(AudioManager.EXTRA_AUDIO_PLUG_STATE, 0) == 1) {
+      supportedEncodings.addAll(Ints.asList(
+          intent.getIntArrayExtra(AudioManager.EXTRA_ENCODINGS)));
       return new AudioCapabilities(
-          Api29.getDirectPlaybackSupportedEncodings(), DEFAULT_MAX_CHANNEL_COUNT);
+          Ints.toArray(supportedEncodings.build()), /* defaultValue= */ DEFAULT_MAX_CHANNEL_COUNT);
```

Comment:
Should we pass `intent.getIntExtra(AudioManager.EXTRA_MAX_CHANNEL_COUNT, /* defaultValue= */ DEFAULT_MAX_CHANNEL_COUNT))` as the `maxChannelCount` in this case?

Reply:
@tianyif This will caused problems with some Xiaomi DTS:X TV as it reports AudioManager.EXTRA_MAX_CHANNEL_COUNT as 8 instead of the correct value of 10 (For layout 5.1.4). This will cause DTS:X profile2 playback to fail.

Reply:
@tianyif I confirm that there are some DTS:X devices that are reporting  ACTION_HDMI_AUDIO_PLUG.EXTRA_MAX_CHANNEL_COUNT as 8 instead of the required 10 channels. 

If the AudioCapabilities is constructed with maxChannelCount==8, it will cause playback to stop in 
Method getEncodingAndChannelConfigForPassthrough()
...
channelCount = getMaxSupportedChannelCountForPassthrough(encoding, sampleRate);
    } else {
      channelCount = format.channelCount;
      **_if (channelCount > maxChannelCount) {_**

This is related to a similar issue about 2 weeks ago. I have unresolved that conversation so that you can see it. Since this is a device issue, how about we proceed with constructing AudioCapabilities with the maxChannlCount from the device as you have suggested. We then handle the channelCount=format.channelCount comparison as a special case and file a bug to come back to this when the devices are fixed (as suggested by @christosts in the earlier conversation that I just unresolved). What do you think? Please see recommended changes here: 78a0385b8d721d0a38947e7ffa0b616e44217983 Thanks.

P/S: I will file the DTS:X Device maxchannelcount reporting bug in Exoplayer and update the comment in the code once you ok this.

Reply:
Acknowledged. I will accept this pull request and move it into the internal review stage, during which our team will discuss how to address the mentioned concern.

Reply:
@tianyif Have filed an issue for this. See https://github.com/androidx/media/issues/396 Comments updated. Please see: 4973af30d3dde53bab5e2dd47b192914765eaa94